### DISCUSSION TITLE
Version 3.14 test results: a list of remarks, questions and change requests

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#762] [General]
First of all:
Congratulations on rolling out a major new version! It looks a lot more organised and I see a lot of solved issues.

I am currently testing the new interface on my staging site. 
I have a list remarks, not sure they are bugs, or that this is just me misinterpreting or not understanding the rationale for changed/new features. I did not find an existing topic about this new version and test results : if there is, feel free to redirect this message.

First off and once more:
I like the clarity of the new interface. I can see you put in a lot of hard work, so kudos for that!
Thank you for working on the import sheet: I re-imported my entire meeting set in the staging site (300+) with the 'delete all' option ticked and it went smoothly and pretty fast.

Here are my test results - in no particular order

**1. Import process: Error codes**
If something in a meeting csv does not work for the plugin, it shows a pretty technical errorcode and it is hit or miss that I can recognise what went wrong. That has not changed from before. But, I want to suggest an improvement.

I did the full import and it referred to the 271th import item and because of the _content_ of the message I knew which one to look for. 

![Screen Shot 2022-04-14 at 12 03 58 PM](https://user-images.githubusercontent.com/59859095/163429069-2acdc87d-46be-49b0-8006-a8da6cfff671.png)

It was a meeting that used double quotes in its meeting notes and that tends to break the import. So that is fine. But I wonder if there is additional information that could be shown. I was not sure whether that meeting was imported or not. It was not. Would it be possible to add to the message that one meeting was skipped? It is not an essential thing, but it would help to know where to look. I realise that you may not know which part of the import failed, just _because_ there was an error in the field matching. But maybe there can be a summary of whatever was there to help us find which meeting to correct.

I still believe that the import could benefit from something in the interface that lets us preview the information we are importing. It could be done before the import starts, or it could show us, at the end of the import, any failing meetings with the field matches, so that we know where to go to correct the import sheet. 

![Screen Shot 2022-04-14 at 12 09 47 PM](https://user-images.githubusercontent.com/59859095/163430043-9073ce86-b29d-4cf3-b622-a99a340ac5f4.png)

**2. Redundancy in meeting day and time in group section details page**
In the meeting details page I feel that the repeating of the meeting day and time in the top of the page, the venue **and** group information seems a bit redundant if the group only has one meeting. I think the information was added to display multiple meetings and that makes sense. But I wonder if it is possible to adjust the code so that it only shows that information if there are more than one meeting? I also wonder if it makes sense to show meetings held at the same venue. In my case (AA may be different), I only have a hand-full of groups that have multiple meetings.

A suggestion for improvement might be to do the same you have done with the open/closed description. 
Make it a button that users can click to see which meetings are held at a particular venue. 
And change the multiple group meetings section with a button: show all meeting days, **if** the group has multiple meetings. Less is more, I think, in the details page, to keep it transparent, compact and clear.
See screenshot: This is for a one meeting group page. The redundant information already takes up a lot of space.

![Screen Shot 2022-04-14 at 10 37 51 AM](https://user-images.githubusercontent.com/59859095/163413612-d0db2a29-382c-4a1e-ae35-68e6c129389a.png)

**3. Performing a search**
I am having some trouble getting consistent results when performing a manual search in the most left filter box.
It seem to be connected to the other boxes. I have 1 meeting that has the prefix '(Phone-only)'. I know that this meeting is on a Wednesday night in Mississauga. When I wanted to see all meetings on that day, I did not find that meeting included. So my next step was to enter 'Phone' in the search box. This did not give me the desired results and I double-checked, in edit mode, if this meeting had been imported. I saw it was there. Went back to the front end and tried a couple of other things.

I think the interface has reverted back to an error/misconception of an older version. So I think maybe the following could be changed:

- If a user used the search box >>> automaticallyempty whatever is there in the other boxes before performing the search. That way it always works. If the other boxes are set to a day and a time of the day, even _if_ that day and time are correct, the search will not work.

![Screen Shot 2022-04-14 at 10 45 36 AM](https://user-images.githubusercontent.com/59859095/163415069-2460ec3c-a44f-4a86-9e63-3fc4c4f900d9.png)

I found this a bit confusing; I tried explicitly clicking 'search' and that also did not give me what I needed. Made me think the meeting was not there, but it was, as you can see in the screenshot.

**4. Sorting after searching**
The interface usually auto sorts by day of the week and meeting time.
However, it does not seem to sort after performing an open search. 
The column labels used to be clickable to perform a sort by any of the columns.
That feature seems to have been lost or removed.
Can we get that back?

It is very helpful to be able to sort if you are in a big set and just not finding the meeting you are trying to find. With sort options it is easier to zoom in on what you are looking for.

**5. Use tooltips**
A lot of people, or so I hear, do not click on things they don't understand. The use of tooltips for buttons might be something that people find helpful.

**6. Search by District instead of region: why was it removed?**
Searching by District was extremely helpful in the old interface. Members checking to see if all meetings are listed use it a lot, especially when they comparing one site to the other. Can this be brought back or have district been removed entirely? I know that district have no real meaning for newcomers, but a lot of members use our website as well and for them searching by district was an essential.

I am also missing a scrollbar in the list when in map-mode. I get as far as Georgetown, the rest of the alphabetised list is not responding to scrolling. It is working find in the regular list mode.

![Screen Shot 2022-04-14 at 11 58 24 AM](https://user-images.githubusercontent.com/59859095/163428417-d63bc6d2-7a12-4a96-ad6a-b18352300321.png)

**7. Show version in 'settings' tab**
Now that we have a tab for settings, it would be useful to show the current version indicator in that tab. Maybe I am looking over it, but I was unable to find it and had to navigate to the plugin page to find it.

**8. Alateen missing from list of programs**
The plugin is marketed as a tool for all 12-step programs. Is it possible to add Alateen to the program choices? It is important to all of Al-Anon to have this program included, even for AA, since this is a program focused on teenage children of alcoholics. If Al-Anon is in the list, Alateen belongs there as well.

**'9. Request a change to this listing' - now: 'Update meeting info button'**
Not happy with this option. 
I liked that it was a short form that would include the meeting essentials for reference, including the meeting notes which, in my case, contain the WSO ID for each group. All of the info that is sent along is not visible (and therefore not distracting) the user who wants to sent a quick remark.

Now that it wants to choose a default email program, it means 2 things:

9.1 -the user can see which email addresses the update is going to.

Which means they can remove someone from the 'send to' list. That is not what I want to happen: we have an Area Group Record Coordinator and myself, the website coordinator. By using the old 'update' form, the info was automatically sent to both coordinators, without the user being able to meddle with the recipients. This was an extremely useful method of keeping both updated without any work for the user.

9.2 -Missing information

I have often asked for a field to log the WSO ID (the only unique identifier of groups in Al-Anon).
There was a good workaround for the missing field: when users clicked the 'request a change' link in the details page, they entered their comments and an excerpt from the meeting they were responding from contained the meeting notes. That was where I was told to put the WSO ID. I did, and it was a tremendous help in keeping the update processes efficient.

That has now been removed. The info sent in the new interface requires me to click on the relevant meeting to find out their WSO ID. If offers a list and a useless comment referring to a row#. That may be useful info in the backend for the development team but it means nothing to me. So that is a step back for me.

Can we talk about bringing the small webform back - see screenshot below? 
Or giving admins the options of using the clickable 'mail to' or the websform? That way admins can follow their own preference.

If neither is possible, then at the very least include some more information in the email, so that we do not have to click a link. Or give us a field for a unique numeric identifier. And include it in the email.
Or, if that is not possible: include the Post ID from the backend.

Why? The link brings us to the front end of the website, to the relevant meeting. To edit, we still have to do a fair bit of work. Go to the dashboard, go to the meetings list, find the meeting, then do the edit. Too much work, not efficient, not an improvement.
In my case, I would only need the WSO ID or the Post ID. I have direct links based on the post ID in the files that I keep off-website for each meeting.

See below the old setup with the webform and below that what the incoming email looks like now.

The old webform:
![Screen Shot 2022-04-14 at 11 41 53 AM](https://user-images.githubusercontent.com/59859095/163425441-4c7c9175-9873-4830-8378-3d463baaf05f.png)

Content of old email notification:
![Screen Shot 2022-04-14 at 12 22 25 PM](https://user-images.githubusercontent.com/59859095/163431967-7e25de8a-d6dc-4d49-92d7-e994258298fa.png)

Content of new email notification (I just typed: 'Meeting feedback", those words are mine. They are not part of the default reply format):
![Screen Shot 2022-04-14 at 12 16 27 PM](https://user-images.githubusercontent.com/59859095/163431045-11067acf-876e-4bcf-9ec9-83dcca5f99cc.png)

That's it for now. Thanks for your consideration and good luck with de-bugging!

Nicky D
Website Coordinator Al-Anon Ontario South

**Comment by joshreisner** [comment]
hi @OSAwebsite - as the maintainer of TSML UI i can answer a few of these items:

* not listing meetings in the sidebar when there's more than one meeting - good idea, thanks. i [created an issue](https://github.com/code4recovery/12-step-meeting-list/discussions/762)
* search: sounds like you're encountering a bug with the filtering. if you could share a URL with the issue i'm happy to look at it. keyword searches should preserve your current filters, they should not be clearing them
* sorting: meetings are always sorted chronologically (number of minutes from now). if that's getting lost somehow after searching then that's a bug and i'm happy to take a look with a URL. sortable columns is not on the development roadmap
* tooltips: not on the roadmap
* districts were removed from the Anywhere dropdown because they are program jargon. however they are still represented on the meeting detail screen, and they should still searchable. you could make a list of them and put them in site navigation, eg `/meetings?search=district+13`
* feedback email: we are working on a major expansion of the form, coming soon, and will have a link to edit the meeting be a part of the message. we are a little bit limited here by the technology in terms of how they can be displayed, but we can address that when the feature is released

**Comment by OSAwebsite** [comment]
Give me a time and a day you want to check it out. The Staging website is
always under construction but I could make it go live when I know when.
Just contact me directly for that info at ***@***.***, if you
don't mind.

Listing of multiple meetings: if you can limit it to groups with multiple
meetings, that would be great. But I would prefer seeing it as an option,
rather than a default, by clicking a button of you are interested in seeing
that information. Being over-complete is not always a blessing.

Search: yes, bug, try it out.

Sorting: columns were sorted, still are in the legacy screen with the new
version. So loosing that option does sound like a bug

Tooltip: nice to have. Not a bug. But helpful to all the non-technical folk
out there

Districts: I try to keep the website as simple as possible - my people do
not take to entering something custom in the site navigation. They might be
considered jargon, but think about this scenario: multiple telephone
helplines use our websites to inform callers. They use that District search
all the time. It just narrows down the area where we can offer meetings to
newcomers. I think it should come back. I really do :)

Feedback email: don't fall in the trap over overcompleteness there. My own
experience with my members is that they loved how it was for the simple
reason that all they had to do was write something like: "meeting back
in-person", without having to dig through a big form. I think it is a great
service to offer. I am not sure AA works the same as Al-Anon, but our
version of your GSO has a lengthy form for more complex changes people wish
to communicate. The Ontario South area has their own version. It is not
hard to fill out, but people simply don't like to fill out forms. The
feedback form kept it nice and simple. Maybe you can work on having 3
settings for admins: a short form (as it was), an email trigger or a bigger
form with check boxes. But I fear that it will overcomplicate something
that worked really well before. if it ain't broken, don't fix it and keep
it simple and more of that kind of good stuff.... :)

My five cents...

Let me know when you want to connect.

Thanks for your quick response!


-
Yours in service,
Nicky D
Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
